extensions/src/test/java/org/pentaho/platform/plugin/services/email/EmailServiceTest.java

```diff
@@ -0,0 +1,479 @@
+    Map<String, String> ret = Arrays.stream( input.split( splitProperty ) )
+            .map( l -> l.split( splitKeyValue )  )
+            .collect( Collectors.toMap(
+              m -> m[0], // key
+              m -> m[1] // value
+            ) );
+    return ret;
+  }
+
+
+  @Test
+  public void testGetOAuthTokenAuthCode_mockHttpServer_httpException() throws Exception {
+    MockWebServer server = new MockWebServer();
+    String filename = "EmailService/EmailAuthenticationResponse1.json";
+    String content = readTestFile( filename );
+
+
+    server.enqueue( new MockResponse()
+            .setResponseCode( HttpStatus.SC_NOT_FOUND )
+            .setBody( content )
+    );
+
+    String urlPathToken = "/common/oauth2/v2.0/token";
+    String mockUrlFulToken = server.url( urlPathToken ).toString();
+
+    EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( true, false, "tes@test.com", "Pentaho Scheduler", "", 25,
+                    EmailConstants.PROTOCOL_GRAPH_API, true, "test@test.com", "", true, true );
+    emailConfigOriginal.setAuthMechanism( EmailConstants.AUTH_TYPE_XOAUTH2 );
+    emailConfigOriginal.setClientId( "cid" );
+    emailConfigOriginal.setClientSecret( "secret" );
+
+    emailConfigOriginal.setTokenUrl( mockUrlFulToken  );
+    emailConfigOriginal.setScope( "https://graph.microsoft.com/.default" );
+    emailConfigOriginal.setGrantType( EmailConstants.GRANT_TYPE_AUTH_CODE );
+
+    emailConfigOriginal.setAuthorizationCode( "testCode" );
+    emailConfigOriginal.setRedirectUri( "uri" );
+
+    try {
+      IEmailAuthenticationResponse actualResult = emailService.getOAuthToken( emailConfigOriginal );
+    } catch ( HttpException he ) {
+      assertTrue( he.getMessage().contains( "Unable to get authorization token" ) );
+    } finally {
+      server.shutdown();
+    }
+
+  }
+
+  @Test
+  public void testSendEmailTestBasicAuth() throws Exception {
+    EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( true, false, "test@pentaho.com", "Pentaho Scheduler", "smtp.com", 25,
+                    EmailConstants.PROTOCOL_SMTP, true, "user", "password", true, true );
+    emailConfigOriginal.setAuthMechanism( EmailConstants.AUTH_TYPE_BASIC );
+
+    emailService.sendEmailTest( emailConfigOriginal );
+
+  }
+
+  @Test
+  public  void testEmailServiceSetup() {
+    try {
+      new EmailService( new File( defaultConfigFile, "cannot.exist" ) );
+      fail( "Exception should be thrown when an invalid EmailService is provided" );
+    } catch ( IllegalArgumentException success ) {
+      //ignore
+    }
+
+    IEmailConfiguration emptyConfig = emailService.getEmailConfig();
+    assertEquals( true, BLANK_CONFIG.equals( emptyConfig ) );
+
+    emailService.setEmailConfig( new EmailConfiguration() );
+
+    emptyConfig = emailService.getEmailConfig();
+    assertEquals( true, BLANK_CONFIG.equals( emptyConfig ) );
+  }
+
```

Comment:
you have logic in EmailService: https://github.com/pentaho/pentaho-platform/blob/c9089a52463b25f0353b4e3a96d8252443370b51/extensions/src/main/java/org/pentaho/platform/plugin/services/email/EmailService.java#L224-L229

that setsRefreshToken in EmailConfiguration and calls some other setters.
Where is the unit test verifying this logic?